include/aws/io/event_loop.h

```diff
@@ -327,17 +345,47 @@ struct aws_event_loop_group *aws_event_loop_group_new(
     void *new_loop_user_data,
     const struct aws_shutdown_callback_options *shutdown_options);
 
+/** Creates an event loop group, with clock, number of loops to manage, the function to call for creating a new
+ * event loop, and also pins all loops to hw threads on the same cpu_group (e.g. NUMA nodes). Note:
+ * If el_count exceeds the number of hw threads in the cpu_group it will be ignored on the assumption that if you
+ * care about NUMA, you don't want hyper-threads doing your IO and you especially don't want IO on a different node.
+ */
+AWS_IO_API
+struct aws_event_loop_group *aws_event_loop_group_new_pinned_to_cpu_group(
+    struct aws_allocator *alloc,
+    aws_io_clock_fn *clock,
```

Comment:
Is it worth making these arguments into options structure?

Reply:
yeah, I thought about that, but then wanted to a.) ease api usage transition, and b.) i don't see the group's options changing. Event-loop was different because we'll likely be changing those for a long time.

Reply:
I'm with Ryan here, we have 4 constructors now, and if any new parameter comes along then we'll have 8. We could add 1 new_with_options() and be futureproof 4ever
```
struct aws_event_loop_group_options {
    aws_io_clock_fn *clock; /* if NULL then default used */
    aws_new_event_loop_fn *new_loop_fn; /* if NULL then default used */
    void *new_loop_user_data;
    uint16_t max_threads; /* if 0 then default used */
    bool pinned_to_cpu_group;
    uint16_t cpu_group; /* only used if pinned_to_cpu_group is true */
    struct aws_shutdown_callback_options;
};
```